net/xia/ppal_ether/main.c

```diff
@@ -222,6 +223,69 @@ static void detach_neigh_to_interface(struct fib_xid_ether_main *mether)
 	ether_interface_put(einterface);
 }
 
+/* Creates a cached hdr and alocates to a mfxid. */
+static int xia_ether_header_cache(const struct fib_xid_ether_main *mfxid,
+				  struct hh_cache *hh, __be16 type)
```

Comment:
1. We don't need parameter `hh` since it is just `mfxid->cached_hdr`. You'll also need to drop the modifier `const` for this change.

2. We don't need parameter `type` because, as the name of this function already hints, it must be `htons(ETH_P_XIP)`.

3. After the changes above, this function can't fail. Change its return type to void.